packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -147,18 +153,27 @@ class AppBar extends StatefulWidget implements PreferredSizeWidget {
        assert(primary != null),
        assert(toolbarOpacity != null),
        assert(bottomOpacity != null),
+       assert(automaticallyImplyLeading != null),
```

Comment:
nit: the order of the asserts should match the order of the arguments (see the style guide - we try to keep the order of everything the same everywhere where the order wouldn't otherwise matter).